### PR TITLE
VBM-118: new auction item layout

### PR DESCRIPTION
**Pull request by natata7**
https://user-images.githubusercontent.com/10462332/188267675-a8af422c-f6dc-481e-8648-c1729ba69a1c.mp4

### CODE REVIEW

File: packages/mobile/src/screens/new-item/components/add-photos/add-photos.tsx [hunk 1]
```diff
@@ -1,28 +1,94 @@
...
-    >
-      <View style={styles.btnContainer}>
-        <ImageIcon size={16} style={styles.icon} />
-        <Text style={[globalStyles.fs14, styles.btnText]}>
-          {t('make_a_post.ADD_PHOTOS')}
-        </Text>
+    <>
+      <View>
+        <ScrollView horizontal showsHorizontalScrollIndicator={false}>
+          {images.map((imageItem, index) => {
```
Comment (seal323) [inline packages/mobile/src/screens/new-item/components/add-photos/add-photos.tsx:75]:
```suggestion
          {images.map(renderImage)}
```

File: packages/mobile/src/screens/new-item/components/add-photos/add-photos.tsx [hunk 2]
```diff
@@ -1,28 +1,94 @@
...
+    const link = await imageService.uploadImage(file);
+
+    setImages((state) => [...state, link]);
+    setIsUploading(false);
+  }, []);
+
+  const deleteImage = (index: number) => {
+    const updateImages = [...images];
+    updateImages.splice(index, 1);
+    setImages(updateImages);
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/add-photos/add-photos.tsx:40]:
```suggestion
    setImages(state => state.filter((_, idx) => idx !== index));
```

File: packages/mobile/src/screens/new-item/components/new-auction-form/new-auction-form.tsx [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+  View,
+  Popover,
+  DateTimePicker,
+  ClockIcon,
+  CalendarIcon,
+} from '~/components/components';
+import { useAppForm, useTranslation, useState } from '~/hooks/hooks';
+import { CALLING_CODE, CITIES, COUNTRIES } from '~/mock/new-item';
+import { globalStyles } from '~/styles/styles';
+import dayjs from 'dayjs';
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-auction-form/new-auction-form.tsx:18]: Pls move it up

File: packages/mobile/src/screens/new-item/components/new-auction-form/new-auction-form.tsx [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+
+import { useStyles } from './styles';
+
+const NewAuctionForm: FC = () => {
+  const { t } = useTranslation();
+  const styles = useStyles();
+
+  const today = new Date();
+  const tomorrow = new Date();
+  tomorrow.setDate(today.getDate() + 1);
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-auction-form/new-auction-form.tsx:30]: I'd recommend using dayjs for date manipulations.
```
const today = dayjs();
const tomorrow = today.add(1, 'day');
...
someFnRequiringDate(tomorrow.toDate())
```

File: packages/mobile/src/screens/new-item/components/new-auction-form/new-auction-form.tsx [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+          is24Hour={true}
+          display="default"
+          minimumDate={tomorrow}
+          onChange={(_, selectedDate) => {
+            setOpenDate(false);
+            const currentDate = dayjs(selectedDate).format('DD/MM/YYYY');
+            setDate(selectedDate ?? tomorrow);
+            setValue('endDate', currentDate);
+          }}
+        />
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-auction-form/new-auction-form.tsx:103]: This should be extracted to a separate controlled component. In that case you won't need to use `setValue` explicitly

File: packages/mobile/src/components/date-time-picker/date-time-picker.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  name,
+  control,
+  mode,
+  label,
+  contentContainerStyle,
+}: Props<T>): ReactElement => {
+  const [open, setOpen] = useState(false);
+  const { field } = useFormControl({ name, control });
+  const styles = useStyles();
+  const { value, onChange } = field;
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/date-time-picker/date-time-picker.tsx:48]:
```suggestion
  const { field: { value, onChange }} = useFormControl({ name, control });
  const styles = useStyles();
```

File: packages/mobile/src/components/date-time-picker/date-time-picker.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+import { useStyles } from './styles';
+
+type Props<T extends FormControlValues> = {
+  placeholder: string;
+  label: string;
+  name: FormControlPath<T>;
+  control: FormControl<T>;
+  errors: FormControlErrors<T>;
+  contentContainerStyle?: StyleProp<ViewStyle>;
+  mode: 'date' | 'time';
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/date-time-picker/date-time-picker.tsx:34]: Let's extract it as a separate enum

File: packages/mobile/src/components/date-time-picker/date-time-picker.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+            style={styles.inputIcon}
+          />
+        )}
+        {mode === 'time' && (
+          <ClockIcon
+            size={22}
+            color={ColorPalette.GRAY_300}
+            style={styles.inputIcon}
+          />
+        )}
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/date-time-picker/date-time-picker.tsx:94]:
```
const Icon = mode === 'time' ? ClockIcon : CalendarIcon;
<Icon size={22} color={ColorPalette.GRAY_300} style={styles.inputIcon} />
```

File: packages/mobile/src/components/date-time-picker/date-time-picker.tsx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+}: Props<T>): ReactElement => {
+  const [open, setOpen] = useState(false);
+  const { field } = useFormControl({ name, control });
+  const styles = useStyles();
+  const { value, onChange } = field;
+
+  const today = dayjs();
+  const tomorrow = today.add(1, 'day');
+  const date = value ? dayjs(value).format('DD/MM/YYYY') : null;
+  const time = value ? dayjs(value).format('HH:mm') : null;
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/date-time-picker/date-time-picker.tsx:53]:
```suggestion
  const date = value ? dayjs(value).format(DateTimeFormat.DATE_ONLY) : null;
  const time = value ? dayjs(value).format(DateTimeFormat.TIME_ONLY) : null;
```

File: packages/mobile/src/screens/new-item/components/add-photos/add-photos.tsx [hunk 10]
```diff
@@ -1,28 +1,90 @@
...
+import { ColorPalette } from '@vse-bude/shared';
 import { styles } from './styles';
 
 const AddPhotos: FC = () => {
   const { t } = useTranslation();
+  const [images, setImages] = useState<string[]>([]);
+  const [, setIsUploading] = useState(false);
+
+  const onGalleryOpen = useCallback(async () => {
+    setIsUploading(true);
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/add-photos/add-photos.tsx:24]: What's the logic behind setting loading state and not using the value?

File: packages/mobile/src/screens/new-item/components/new-auction-form/new-auction-form.tsx [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+        control={control}
+        errors={errors}
+        placeholder={'-/-/-'}
+        mode={'date'}
+        contentContainerStyle={globalStyles.mt5}
+      />
+
+      <DatePicker
+        label={t('make_a_post.ENDING_TIME')}
+        name="endDate"
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-auction-form/new-auction-form.tsx:71]: You have the same name for both pickers.

**Review by AleksandrTermenzhy — COMMENTED** [review]
Also resolve conflicts pls

**Review by AleksandrTermenzhy — APPROVED** [review]

**Review by seal323 — APPROVED** [review]

**Pull request merged**